src/Features/Blockcore.Features.Consensus/Rules/UtxosetRules/CheckPosUtxosetRule.cs

```diff
@@ -210,5 +210,14 @@ public virtual Money GetProofOfStakeReward(int height)
 
             return this.consensus.ProofOfStakeReward;
         }
+
+        /// <summary>
+        /// If the block can have no rewards.
+        /// </summary>
+        /// <returns>True - By default require a reward for new blocks</returns>
+        public virtual bool RewardRequired()
```

Comment:
I don't think this is needed once fee is zero in options

Reply:
It still needs to be here because reward <= 0 will reject the block otherwise.

Reply:
Ah I wonder if that condition is correct ` if (reward <= 0)`